Source/Controllers/Other/SecureBookmark.swift

```diff
@@ -0,0 +1,69 @@
+//
+//  SecureBookmark.swift
+//  Sequel Ace
+//
+//  Created by James on 7/12/2020.
+//  Copyright © 2020 Sequel-Ace. All rights reserved.
+//
+
+import Foundation
+import os.log
+import Firebase
+
+final class SecureBookmark: NSObject {
+    private let bookmarkData: SecureBookmarkData
+
+    static var emptySucureBookmarkData: SecureBookmarkData{
+        return SecureBookmarkData(data: Data(), options: 0, url: URL(string: "nil")!)
+    }
+
+    init(data: Data, options: Double, url: URL) {
+        bookmarkData = SecureBookmarkData(data: data, options: options, url: url)
+        super.init()
+    }
+
+    func getEncodedData() -> Data {
+        if #available(OSX 10.13, *) {
+            do{
+                let codedData = try NSKeyedArchiver.archivedData(withRootObject: bookmarkData, requiringSecureCoding: true)
+                return codedData
+            }
+            catch{
+                os_log("Failed to encode data, Error: %@", log: OSLog.`default`, type: .error, error.localizedDescription)
+                Crashlytics.crashlytics().log("Failed to encode data, Error: \(error.localizedDescription)")
+                return Data() // hmmmmmmm JCS.
```

Comment:
if we could return `optional` (`Data?`), we could return here nil instead of this

Reply:
How does that help though? We'd need to unwrap when we use the function

Reply:
Do we want to exit if bookmark data is nil? I guess so

Reply:
put it in a guard?

```
guard let spData = sp.getEncodedData() else {
                return <#default value#>
                
            }
```

Reply:
Exactly - returning empty data is "valid data" for calling functions, where unwrapping nil value in the calling function tells the function "something is wrong, do what you need"